### PR TITLE
feat(cbor): Add support for CBOR encoding

### PR DESCRIPTION
**Pull request by imsk17**
Required for implementation of [3156](https://github.com/gofiber/fiber/pull/3156)

## Summary by CodeRabbit

- **New Features**
	- Introduced functions for encoding and decoding CBOR data in the utils package, enhancing data handling capabilities.

- **Tests**
	- Added new tests for verifying the functionality of CBOR encoding and decoding processes, ensuring reliability and correctness.

### CODE REVIEW

**Comment by gaby** [comment]
@imsk17 This is the wrong repo? It should be https://github.com/gofiber/fiber

**Comment by gaby** [comment]
@imsk17 Closing this. The implementation is for gofiber/fiber like your previous PR. It just had the wrong target branch.

Please open the PR in https://github.com/gofiber/fiber

**Comment by imsk17** [comment]
But that repo depends on utils, which is where I have to add the CBOR format first like all other formats.

**Comment by gaby** [comment]
> But that repo depends on utils, which is where I have to add the CBOR format first like all other formats.

I see whatbyou mean, makes sense 💪 PR has been re-opened

**Comment by gaby** [comment]
@imsk17 Can you take a look at some of the lint failures and suggestions.

With the makefile you can run `make lint` locally

**Comment by imsk17** [comment]
Done.

**Comment by gaby** [comment]
@coderabbitai full review

**Comment by gaby** [comment]
> Done.

Awesome, I will review today

**Comment by imsk17** [comment]
Can we get this to merge?

File: cbor_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	var (
+		ss                 sampleStructure
+		importantString, _               = hex.DecodeString("a170696d706f7274616e745f737472696e676b48656c6c6f20576f726c64")
+		cborDecoder        CBORUnmarshal = cbor.Unmarshal
+	)
+
+	err := cborDecoder(importantString, &ss)
+	require.NoError(t, err)
+	require.Equal(t, "Hello World", ss.ImportantString)
+}
```
[the remark this answers is not in the document]
Comment (gaby) [reply cbor_test.go]: @imsk17 Can you add other tests as suggested here. For example:
- Empty structure
- Bad input
- Nested structure

**Review by gaby — CHANGES_REQUESTED** [review]
Overall looks good, just needs more tests. Unlike json/xml that are built into Golang, this relies on external packages so we should add a few more basic tests

**Comment by gaby** [comment]
@imsk17 Can you add the missing unit-tests? Thanks

**Comment by imsk17** [comment]
On it.

**Comment by imsk17** [comment]
Let me know of more test cases if these arent enough.

**Comment by gaby** [comment]
> Let me know of more test cases if these arent enough.

Golangci-lint doesnt seem to like the comparison against the same variable

**Review by gaby — APPROVED** [review]
👍 LGTM

**Comment by gaby** [comment]
@imsk17 This will be merged once someone else in the team reviews it. Thanks! 💪

**Review by ReneWerner87 — APPROVED** [review]

**Comment by gaby** [comment]
@imsk17 CBOR support is now in the latest gofiber/utils release.

https://github.com/gofiber/utils/releases/tag/v2.0.0-beta.7

**Pull request merged**